packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -476,7 +476,10 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
 
   /** @private */
   __dispatchChange() {
-    this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+    if (this.value !== this._lastCommittedValue) {
```

Comment:
I believe we also need to update `_lastCommittedValue` on programmatic value change.

1. The value is set to `10:00` programmatically.
2. The user tries to commit `10:0`
3. `10:0` is automatically corrected to `10:00`

**Result:** Currently, it will result in a `change` event because `_lastCommittedValue` was `undefined.

Reply:
Good point, thanks. I will update the logic to cover this case.